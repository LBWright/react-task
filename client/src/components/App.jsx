import React, { Component } from 'react'
import io from 'socket.io-client'
const URL = 'http://localhost:8000'

class App extends Component {
  state = {
    socket: null,
  }

  componentWillMount() {
    this.initializeConnection()
  }

  initializeConnection = () => {
    const socket = io(URL)
    try {
      socket.on('connect', () => {
        console.log('connected')
      })
      this.setState({ socket })
    } catch (e) {
      console.log(e)
    }
  }

  render() {
    this.state.socket.on('data', data => {
      console.log(data)
    })
    return <div>Practical Intro To WebSockets.</div>
  }
}

export default App
