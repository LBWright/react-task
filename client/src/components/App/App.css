.container {
  display: flex;
}
.warning-toast {
  color: black;
}
