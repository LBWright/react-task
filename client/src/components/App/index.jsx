import React, { useState } from 'react'
import useThresholdAlert from '../../hooks/useThresholdAlert'
import useWebSocket from '../../hooks/useWebSocket'
import { Chart, Threshold, Toast } from '..'

import './App.css'
// Working with CSS to keep consistency with react-toastify
// prefer CSS-in-JS or preprocessor - would refactor

const URL = 'http://localhost:8000'

// use of function keyword preferred over arrow func in functional components
// faster in most cases - doesn't matter for closure.
function App() {
  const [threshold, setThreshold] = useState()

  const { data, currentData, error } = useWebSocket(URL)
  useThresholdAlert(currentData, threshold)
  return (
    <div className="container">
      <Chart data={data} />
      <Threshold setThreshold={setThreshold} threshold={threshold} />
      <Toast />
    </div>
  )
}

export default App
