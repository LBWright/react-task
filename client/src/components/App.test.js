import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import App from './App'

describe('<App />', () => {
  const app = mount(<App />)
  it('should render without crashing', () => {
    expect(app).toBeTruthy()
  })
  it('should render <Chart />', () => {
    expect(app.find('Chart')).toBeTruthy()
  })
  it('should render <Threshold />', () => {
    expect(app.find('Threshold')).toBeTruthy()
  })
  it('should render <Toast />', () => {
    expect(app.find('Toast')).toBeTruthy()
  })
  it('should match snapshot', () => {
    const appSnap = renderer.create(<App />)
    expect(appSnap.toJSON()).toMatchSnapshot()
  })
})
