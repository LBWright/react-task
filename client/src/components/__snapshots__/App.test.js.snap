// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div
  className="container"
>
  <div
    className="recharts-wrapper"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "cursor": "default",
        "height": 800,
        "position": "relative",
        "width": 1080,
      }
    }
  >
    <svg
      className="recharts-surface"
      height={800}
      version="1.1"
      viewBox="0 0 1080 800"
      width={1080}
    >
      <defs>
        <clipPath
          id="recharts5-clip"
        >
          <rect
            height={730}
            width={980}
            x={80}
            y={20}
          />
        </clipPath>
      </defs>
      <g
        className="recharts-cartesian-grid"
      >
        <g
          className="recharts-cartesian-grid-horizontal"
        >
          <line
            fill="none"
            height={730}
            stroke="#ccc"
            width={980}
            x={80}
            x1={80}
            x2={1060}
            y={20}
            y1={20}
            y2={20}
          />
          <line
            fill="none"
            height={730}
            stroke="#ccc"
            width={980}
            x={80}
            x1={80}
            x2={1060}
            y={20}
            y1={750}
            y2={750}
          />
        </g>
        <g
          className="recharts-cartesian-grid-vertical"
        >
          <line
            fill="none"
            height={730}
            stroke="#ccc"
            width={980}
            x={80}
            x1={80}
            x2={80}
            y={20}
            y1={20}
            y2={750}
          />
          <line
            fill="none"
            height={730}
            stroke="#ccc"
            width={980}
            x={80}
            x1={1060}
            x2={1060}
            y={20}
            y1={20}
            y2={750}
          />
        </g>
      </g>
      <g
        className="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
      >
        <line
          className="recharts-cartesian-axis-line"
          fill="none"
          height={730}
          stroke="#666"
          width={60}
          x={20}
          x1={80}
          x2={80}
          y={20}
          y1={20}
          y2={750}
        />
        <g
          className="recharts-cartesian-axis-ticks"
        />
      </g>
    </svg>
    <div
      className="recharts-legend-wrapper"
      style={
        Object {
          "bottom": 20,
          "height": "auto",
          "left": 20,
          "position": "absolute",
          "width": 1040,
        }
      }
    >
      <ul
        className="recharts-default-legend"
        style={
          Object {
            "margin": 0,
            "padding": 0,
            "textAlign": "center",
          }
        }
      >
        <li
          className="recharts-legend-item legend-item-0"
          style={
            Object {
              "display": "inline-block",
              "marginRight": 10,
            }
          }
        >
          <svg
            className="recharts-surface"
            height={14}
            style={
              Object {
                "display": "inline-block",
                "marginRight": 4,
                "verticalAlign": "middle",
              }
            }
            version="1.1"
            viewBox="0 0 32 32"
            width={14}
          >
            <path
              className="recharts-legend-icon"
              d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
              fill="none"
              stroke="#8884d8"
              strokeWidth={4}
            />
          </svg>
          <span
            className="recharts-legend-item-text"
          >
            value
          </span>
        </li>
      </ul>
    </div>
    <div
      className="recharts-tooltip-wrapper"
      style={
        Object {
          "MozTransform": "translate(undefinedpx, undefinedpx)",
          "OTransform": "translate(undefinedpx, undefinedpx)",
          "WebkitTransform": "translate(undefinedpx, undefinedpx)",
          "msTransform": "translate(undefinedpx, undefinedpx)",
          "pointerEvents": "none",
          "position": "absolute",
          "top": 0,
          "transform": "translate(undefinedpx, undefinedpx)",
          "visibility": "hidden",
        }
      }
    >
      <div
        className="recharts-default-tooltip"
        style={
          Object {
            "backgroundColor": "#fff",
            "border": "1px solid #ccc",
            "margin": 0,
            "padding": 10,
            "whiteSpace": "nowrap",
          }
        }
      >
        <p
          className="recharts-tooltip-label"
          style={
            Object {
              "margin": 0,
            }
          }
        >
          
        </p>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "margin": 20,
      }
    }
  >
    <h2>
      Threshold Controls
    </h2>
    <form
      id="threshold-form"
      onSubmit={[Function]}
    >
      <input
        id="threshold-input"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Set Alert Threshold"
        type="number"
        value=""
      />
      <button
        id="threshold-submit"
        onClick={[Function]}
        type="submit"
      >
        Set Threshold
      </button>
    </form>
    <div>
      <p>
        No Alert Threshold Specified
      </p>
    </div>
  </div>
  <div
    className="Toastify"
  />
</div>
`;
